docker/common/install_mpi4py.sh

```diff
@@ -17,56 +17,54 @@ trap 'rm -rf "$TMP_DIR"' EXIT
 # Download and extract in one step
 curl -L ${RELEASE_URL} | tar -zx -C "$TMP_DIR"
 
-# Bypassing compatibility issues with higher versions (>= 69) of setuptools.
-sed -i 's/>= 40\.9\.0/>= 40.9.0, < 69/g' "$TMP_DIR/mpi4py-${MPI4PY_VERSION}/pyproject.toml"
-
 # Apply the patch
 cd "$TMP_DIR/mpi4py-${MPI4PY_VERSION}"
 git apply <<EOF
-diff --git a/src/mpi4py/futures/_lib.py b/src/mpi4py/futures/_lib.py
-index f14934d1..eebfb8fc 100644
---- a/src/mpi4py/futures/_lib.py
-+++ b/src/mpi4py/futures/_lib.py
-@@ -278,6 +278,40 @@ def _manager_comm(pool, options, comm, full=True):
-
-
- def _manager_split(pool, options, comm, root):
-+    if(os.getenv("TRTLLM_USE_MPI_KVCACHE")=="1"):
-+        from cuda import cudart
-+        has_slurm_rank=False
-+        has_ompi_rank=False
-+        slurm_rank=0
-+        ompi_rank=0
-+        if(os.getenv("SLURM_PROCID")):
-+            slurm_rank = int(os.environ["SLURM_PROCID"])
-+            has_slurm_rank=True
-+        elif(os.getenv("OMPI_COMM_WORLD_RANK")):
-+            ompi_rank = int(os.environ["OMPI_COMM_WORLD_RANK"])
-+            has_ompi_rank=True
-+        else:
-+            raise RuntimeError("No SLURM_PROCID or OMPI_COMM_WORLD_RANK environment variable found When TRTLLM_USE_MPI_KVCACHE is set to 1")
-+        if(has_slurm_rank and has_ompi_rank):
-+            if(slurm_rank>0 and ompi_rank>0):
-+                raise RuntimeError("Only one of SLURM_PROCID or OMPI_COMM_WORLD_RANK should >0 when TRTLLM_USE_MPI_KVCACHE is set to 1")
+diff --git a/src/mpi4py/futures/_core.py b/src/mpi4py/futures/_core.py
```

Comment:
Is this patch still necessary now considering we already deprecate the mpi-based kv-cache transfer?  @chuangz0